mi/dataset/parser/wavss_a_dcl.py

```diff
@@ -165,17 +65,84 @@ class ArrayParticleKeys(BaseEnum):
 
 
 class WavssADclCommonDataParticle(DataParticle):
+    """
+    WAVSS common DCL particle class - basis for each WAVSS particle
+    """
+
+    # particle maps consists of tuples containing the name of parameter and encoding type
+    # common particle map used by non-directional, directional, and fourier
+    band_parameter_types = [
+        ('number_bands', int),
+        ('initial_frequency', float),
+        ('frequency_spacing', float)
+    ]
+
+    def __init__(self, raw_data,
+                 port_timestamp=None,
+                 internal_timestamp=None,
+                 preferred_timestamp=DataParticleKey.PORT_TIMESTAMP,
+                 quality_flag=DataParticleValue.OK,
+                 new_sequence=None):
+
+        self.dcl_data = None
+
+        super(WavssADclCommonDataParticle, self).__init__(
+            raw_data, port_timestamp, internal_timestamp, preferred_timestamp, quality_flag, new_sequence)
+
+    def check_sum(self):
+        """
+        Extract the data segment and checksum from the raw data string, then make sure that
+        checksum matches provided value.
+        :return: True if the current particle data matches provided checksum (after the '*').
+        """
+        parts = self.raw_data.split(None, 2)
+        if len(parts) == 3:  # standard format with date and time leaders
+            _, _, parts = parts
+        else:
+            parts = self.raw_data
+        if parts[0] != '$':  # data segment must begin with $ leader
+            return False
+        parts = parts.rsplit('*', 1)
+        if len(parts) != 2:  # no checksum delimiter - cannot process
+            return False
+        self.dcl_data, chk_sum = parts
```

Comment:
self.dcl_data is created in check_sum function... ewww

consider two methods, one to find the DCL data and checksum value, one to validate.